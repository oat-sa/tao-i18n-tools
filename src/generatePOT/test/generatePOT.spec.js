/**
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; under version 2
 * of the License (non-upgradable).
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 *
 * Copyright (c) 2020 (original work) Open Assessment Technologies SA
 *
 */

const generatePOT = require('../generatePOT.js');

let strings;

beforeAll(() => {
    strings = new Map([
        ['All work is done!', [{ file: 'index.js', line: 13 }]],
        [
            'You have no more pending tasks for now!',
            [
                { file: '../src/controller/runner.js', line: 31 },
                { file: '../src/controller/runner.js', line: 45 }
            ]
        ],
        ['Logout', [{ file: '../../src/bootstrap.js', line: 22 }]]
    ]);
});

describe('API', () => {
    it('should exist', function () {
        expect(generatePOT).toBeDefined;
    });

    it('should be a function', function () {
        expect(generatePOT).toBeInstanceOf(Function);
    });
});

describe('generatePOT', () => {
    it('should return right POT content', function () {
        const generatePotContent = generatePOT(strings);

        expect(generatePotContent).toMatchSnapshot();
    });
});
