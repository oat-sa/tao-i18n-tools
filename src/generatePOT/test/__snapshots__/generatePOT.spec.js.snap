// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generatePOT should return right POT content 1`] = `
"#: index.js:13
msgid \\"All work is done!\\"
msgstr \\"\\"
#: ../src/controller/runner.js:31
#: ../src/controller/runner.js:45
msgid \\"You have no more pending tasks for now!\\"
msgstr \\"\\"
#: ../../src/bootstrap.js:22
msgid \\"Logout\\"
msgstr \\"\\"
#: ../../src/bootstrap.js:12
msgid \\"Remove test\\"
msgid_plural \\"Remove tests\\"
msgstr[0] \\"\\"
#: ../../src/bootstrap-duplicates.js:123
msgid \\"Remove test\\"
msgid_plural \\"Remove the tests\\"
msgstr[0] \\"\\"
#: ../../src/bootstrap-duplicates.js:1234
msgid \\"Remove a test\\"
msgid_plural \\"Remove tests\\"
msgstr[0] \\"\\"
"
`;
